packages/form-core/src/types.ts

```diff
@@ -2,13 +2,21 @@ import { type DeepKeys } from './util-types'
 
 export type ValidationError = undefined | false | null | string
 
+export type APITypes = 'form' | 'field'
+
 /**
  * If/when TypeScript supports higher-kinded types, this should not be `unknown` anymore
  * @private
  */
 export type Validator<Type, Fn = unknown> = () => {
-  validate(options: { value: Type }, fn: Fn): ValidationError
-  validateAsync(options: { value: Type }, fn: Fn): Promise<ValidationError>
+  validate(
+    options: { value: Type; api: APITypes },
```

Comment:
I find `api: APITypes` too generic (apis of what?), maybe something like `validationSource` or `source`?

The idea of passing that value here is great btw :)

Reply:
I like `validationSource`!